src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -1956,15 +1959,29 @@ CitusHasBeenLoadedInternal(void)
 		/*
 		 * We do not use Citus hooks during CREATE/ALTER EXTENSION citus
 		 * since the objects used by the C code might be not be there yet.
+		 * We set creatingCitus to true in case of a rollback so that we clear
+		 * MetadataCache
 		 */
+		creatingCitus = true;
 		return false;
```

Comment:
`CachedRelationNamespaceLookupExtended()` (or maybe `InitializeCaches()`) might be a better place for this logic. And we can name the variable something more specific, like `cachedDuringCitusCreation`.

Also, it's not critical that we know we are creating the citus extension specifically. So if we're doing this check to set the flag in a hot path, we can just check `creating_extension`. I doubt the creation of another extension would cause the caching of a citus oid (though maybe in a transaction that creates two extensions?), and even if it did in some rare case, it would just be a spurious invalidation which is still correct.